### PR TITLE
refactor(test): add unit test for crane clients

### PR DESCRIPTION
**Pull request by SilverTux**
This PR also aims to achieve https://github.com/dyrector-io/dyrectorio/issues/241 goals testing crane/k8s/clients.go

### CODE REVIEW

**Review by nandor-magyar — CHANGES_REQUESTED** [review]

File: agent/pkg/crane/k8s/clients.go [hunk 1]
```diff
@@ -13,6 +13,11 @@ import (
 	"github.com/dyrector-io/dyrectorio/agent/pkg/crane/config"
 )
 
+var (
+	InClusterConfig      = rest.InClusterConfig
```
[the remark this answers is not in the document]
Comment (SilverTux) [reply agent/pkg/crane/k8s/clients.go]: I've chosen to make a new type where the dependent functions are fields of the struct so I can get rid of using globals, and relatively easy to replace those in unit tests. What do you think about this solution?

**Review by polaroi8d — APPROVED** [review]
LGTM (informal) anyway good job, thanks! ✋🏻

File: agent/pkg/crane/k8s/clients.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nandor-magyar) [reply agent/pkg/crane/k8s/clients.go]: This also looks a bit wonky. The struct has the functions with the same header and a slightly different behavior. However, I don't have any significantly better idea, so still good job :+1:

**Review by nandor-magyar — CHANGES_REQUESTED** [review]

File: agent/pkg/crane/k8s/clients_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	client.InClusterConfig = func() (*rest.Config, error) {
+		callCount = callCount + 1
+		return &rest.Config{}, nil
+	}
+
+	// WHEN
+	resultClient, _ := client.GetClientSet(cfg)
+
+	// THEN
+	assert.NotEqual(t, nil, resultClient)
```
[the remark this answers is not in the document]
Comment (SilverTux) [reply agent/pkg/crane/k8s/clients_test.go]: You are right using that is more elegant, thank you! I've fixed those :)

**Review by m8vago — APPROVED** [review]

**Review by nandor-magyar — APPROVED** [review]

**Pull request merged**